inference-engine/src/inference_engine/include/openvino/runtime/executable_network.hpp

```diff
@@ -0,0 +1,141 @@
+     * object is destroyed.
+     * @param impl Initialized shared pointer
+     */
+    ExecutableNetwork(const std::shared_ptr<SharedObject>& so,
+                      const std::shared_ptr<ie::IExecutableNetworkInternal>& impl);
+    friend class ov::runtime::Core;
+
+public:
+    /**
+     * @brief A default constructor.
+     */
+    ExecutableNetwork() = default;
+
+    /**
+     * @brief Creates an inference request object used to infer the network.
+     *
+     * The created request has allocated input and output blobs (that can be changed later).
+     *
+     * @return InferRequest object
+     */
+    InferRequest create_infer_request();
+
+    /**
+     * @brief Exports the current executable network.
+     *
+     * @see Core::ImportNetwork
+     *
+     * @param modelFileName Full path to the location of the exported file
+     */
+    void export_model(const std::string& modelFileName);
+
+    /**
+     * @brief Exports the current executable network.
+     *
+     * @see Core::ImportNetwork
+     *
+     * @param networkModel Network model output stream
+     */
+    void export_model(std::ostream& networkModel);
+
+    /**
+     * @brief Get executable graph information from a device
+     *
+     * @return Function containing Executable Graph Info
+     */
+    std::shared_ptr<const Function> get_function() const;
+
+    /**
+     * @brief Sets configuration for current executable network
+     *
+     * @param config Map of pairs: (config parameter name, config parameter value)
+     */
+    void set_config(const ie::ParamMap& config);
+
+    /** @brief Gets configuration for current executable network.
+     *
+     * The method is responsible to extract information
+     * which affects executable network execution. The list of supported configuration values can be extracted via
+     * ExecutableNetwork::get_metric with the SUPPORTED_CONFIG_KEYS key, but some of these keys cannot be changed
+     * dynamically, e.g. DEVICE_ID cannot changed if an executable network has already been compiled for particular
+     * device.
+     *
+     * @param name config key, can be found in ie_plugin_config.hpp
+     * @return Configuration parameter value
+     */
+    ie::Parameter get_config(const std::string& name) const;
+
+    /**
+     * @brief Gets general runtime metric for an executable network.
+     *
+     * It can be network name, actual device ID on
+     * which executable network is running or all other properties which cannot be changed dynamically.
+     *
+     * @param name metric name to request
+     * @return Metric parameter value
+     */
+    ie::Parameter get_metric(const std::string& name) const;
+
+    /**
+     * @brief Returns pointer to plugin-specific shared context
+     * on remote accelerator device that was used to create this ExecutableNetwork
+     * @return A context
+     */
+    std::shared_ptr<ie::RemoteContext> get_context() const;
```

Comment:
why is it from ie namespace, we have it in `ov::runtime`. See https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/inference_engine/include/openvino/runtime/remote_context.hpp

Reply:
Will be fixed in https://github.com/openvinotoolkit/openvino/pull/7259 with alongside with core